var/spack/repos/builtin/packages/nccl/package.py

```diff
@@ -17,6 +17,8 @@ class Nccl(MakefilePackage, CudaPackage):
     maintainers = ["adamjstewart"]
     libraries = ["libnccl.so"]
 
+    version("2.16.2-1", sha256="7f7c738511a8876403fc574d13d48e7c250d934d755598d82e14bab12236fc64")
```

Comment:
I am not getting this sha256.  Can you confirm:

```
$ spack checksum nccl@2.16.2-1
==> Found 1 version of nccl:
  
  2.16.2-1  https://github.com/NVIDIA/nccl/archive/v2.16.2-1.tar.gz

==> Fetching https://github.com/NVIDIA/nccl/archive/v2.16.2-1.tar.gz

    version("2.16.2-1", sha256="ba93b52fb7c63d472ad0d2853e8057f1c82ae190cad7c5cac398a64194ea1e57")
```

Reply:
GitHub recently made a change that caused all archive tarballs to be regenerated, so that's probably why you had a different checksum originally. Most of the older existing versions have incorrect checksums now. We'll try to figure out a solution to that for all packages, not just nccl.

@tgamblin

Reply:
Apparently GitHub is reverting this change, so the checksums may have returned to the way they were before. Can you check?

Reply:
GitHub *did* revert the hash algorithm change so the versions should reverted/restored:

```
spack checksum nccl 2.16.2-1 2.15.5-1
==> Found 2 versions of nccl:
  
  2.16.2-1  https://github.com/NVIDIA/nccl/archive/v2.16.2-1.tar.gz
  2.15.5-1  https://github.com/NVIDIA/nccl/archive/v2.15.5-1.tar.gz

==> Fetching https://github.com/NVIDIA/nccl/archive/v2.16.2-1.tar.gz
==> Fetching https://github.com/NVIDIA/nccl/archive/v2.15.5-1.tar.gz

    version("2.16.2-1", sha256="7f7c738511a8876403fc574d13d48e7c250d934d755598d82e14bab12236fc64")
    version("2.15.5-1", sha256="f4ac3c74d469c9cd718f82e1477759785db9b9f8cc9d9ecc103485805b8394a3")
```

Reply:
Hope I get this right, I'm a bit confused

Reply:
Apologies for the confusion.  I wasn't aware of the GitHub algorithm change yesterday.

Reply:
No problem, thanks :)